src/buffer/out/textBuffer.cpp

```diff
@@ -1914,18 +1923,32 @@ HRESULT TextBuffer::Reflow(TextBuffer& oldBuffer, TextBuffer& newBuffer)
     // place the new cursor back on the equivalent character in
     // the new buffer.
     const COORD cOldCursorPos = oldCursor.GetPosition();
-    const COORD cOldLastChar = oldBuffer.GetLastNonSpaceCharacter();
+    const COORD cOldLastChar = lastCharacterViewport.has_value() ?
+                                   oldBuffer.GetLastNonSpaceCharacter(lastCharacterViewport.value()) :
+                                   oldBuffer.GetLastNonSpaceCharacter();

```

Comment:
Any reason to not just make `GetLastNonSpaceCharacter` accept the optional viewport straight up?

Reply:
Not particularly no

Reply:
oh wait ew, `GetLastNonSpaceCharacter` doesn't accept an optional currently. That'll take some finagling